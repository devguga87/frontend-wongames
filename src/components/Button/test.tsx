import { screen } from '@testing-library/react';
import { Button } from '.';
import { renderWithTheme } from '../../utils/tests/helpers';
import { AddShoppingCart } from '@styled-icons/material-outlined/AddShoppingCart';
describe('<Button />', () => {
  it('should render the small size', () => {
    const { container } = renderWithTheme(<Button size="small">Hello</Button>);
    expect(screen.getByRole('button', { name: 'Hello' })).toHaveStyle({
      height: '3rem',
      'font-size': '1.2rem',
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render a button medium size by default', () => {
    renderWithTheme(<Button>Hello</Button>);
    expect(screen.getByRole('button', { name: 'Hello' })).toHaveStyle({
      height: '4rem',
      padding: '0.8rem 3.2rem',
      'font-size': '1.4rem',
    });
  });

  it('should render the large size', () => {
    renderWithTheme(<Button size="large">Hello</Button>);
    expect(screen.getByRole('button', { name: 'Hello' })).toHaveStyle({
      height: '5rem',
      padding: '0.8rem 4.8rem',
      'font-size': '1.6rem',
    });
  });

  it('should render full width version', () => {
    renderWithTheme(<Button fullWidth>Hello</Button>);
    expect(screen.getByRole('button', { name: 'Hello' })).toHaveStyle({
      width: '100%',
    });
  });

  it('should render an icon version', () => {
    renderWithTheme(
      <Button icon={<AddShoppingCart data-testid="icon" />}>Hello</Button>,
    );
    expect(screen.getByRole('button', { name: 'Hello' })).toBeInTheDocument();
    expect(screen.getByTestId('icon')).toBeInTheDocument();
  });
});
