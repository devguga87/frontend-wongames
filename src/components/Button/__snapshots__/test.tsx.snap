// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render the small size 1`] = `
.c0 {
  background: linear-gradient(180deg,#ff5f5f 0%,#f062c0 50%);
  color: #FAFAFA;
  border: 0;
  border-radius: 0.4rem;
  padding: 0.8rem;
  height: 3rem;
  font-size: 1.2rem;
}

<button
  class="c0"
>
  <span>
    Hello
  </span>
</button>
`;
